### PR TITLE
A faulty Content-Type should return the 415 status code, not 406

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/EventListener/DeserializeListener.php [hunk 1]
```diff
@@ -113,7 +114,7 @@ private function getFormat(Request $request): string
          */
         $contentType = $request->headers->get('CONTENT_TYPE');
         if (null === $contentType) {
-            throw new NotAcceptableHttpException('The "Content-Type" header must exist.');
+            throw new UnsupportedMediaTypeHttpException('The "Content-Type" header must exist.');
```
Comment (teohhanhui) [inline src/EventListener/DeserializeListener.php]: https://github.com/api-platform/core/pull/2892/files#diff-0170cc702a42034d94c5d83ccadad1deR108 needs to be updated too.

Comment (dunglas) [reply src/EventListener/DeserializeListener.php]: The link looks bad. Can you edit?

Comment (teohhanhui) [reply src/EventListener/DeserializeListener.php]: The phpdoc:

```diff
-                  * @throws NotAcceptableHttpException
+                  * @throws UnsupportedMediaTypeHttpException
```

Comment (dunglas) [reply src/EventListener/DeserializeListener.php]: got it, thanks

**Comment by teohhanhui** [comment]
Thanks @dunglas! 🚀

**Pull request merged**